chore: TypeScript v5→v6 メジャーアップデート対応

## 概要

Dependabot PR #362 (\`typescript\` 5.9.3 → 6.0.2) が \`npm ci\` の ERESOLVE で CI 失敗。複数の peer dependency が TypeScript 6 を未サポート。

## CI 失敗内容

\`\`\`
npm error ERESOLVE could not resolve
npm error While resolving: @sveltejs/kit@2.55.0
npm error Found: typescript@6.0.2
npm error Could not resolve dependency:
npm error peerOptional typescript@"^5.3.3" from @sveltejs/kit@2.55.0
\`\`\`

**非互換の peer dependency:**
- \`@sveltejs/kit\`: peerOptional \`typescript@^5.3.3\`
- \`@typescript-eslint/parser\`: peer \`typescript@>=4.8.4 <6.1.0\`（これは OK）

## 対応方針

SvelteKit が TypeScript 6 をサポートするバージョンをリリースするまで待つ必要がある。\`--legacy-peer-deps\` での強制インストールは本番環境のリスクが高く非推奨。

## ゴール

- [ ] SvelteKit の TypeScript 6 サポート状況を確認
- [ ] SvelteKit が TS6 対応したタイミングで再度アップデートを実施
- [ ] 全 peer dependency の TS6 互換性を確認
- [ ] CI 全パス確認

## 関連

- PR #362: https://github.com/Takenori-Kusaka/ganbari-quest/pull/362

--- a/package.json
+++ b/package.json
@@ -76,7 +76,7 @@
 		"svelte-eslint-parser": "^1.6.0",
 		"tailwindcss": "^4.2.2",
 		"tsx": "^4.19.0",
-		"typescript": "^5.9.3",
+		"typescript": "^6.0.2",
 		"vite": "^8.0.3",
 		"vitest": "^4.1.2"
 	},

